packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -270,31 +279,32 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room, userId) {
 			RocketChat.Sandstorm.notify(message, [userOfMention._id],
 				`@${ user.username }: ${ message.msg }`, 'privateMessage');
 
-		}
-		if ((userOfMention != null) && canBeNotified(userOfMentionId, 'mobile')) {
-			const duration = settings.desktopNotificationDurations[userOfMention._id];
-			notifyDesktopUser(userOfMention._id, user, message, room, duration);
-		}
+			if (canBeNotified(userOfMentionId, 'desktop')) {
+				const duration = settings.desktopNotificationDurations[userOfMention._id];
+				notifyDesktopUser(userOfMention._id, user, message, room, duration);
+			}
 
-		if ((userOfMention != null) && canBeNotified(userOfMentionId, 'desktop')) {
-			if (Push.enabled === true && userOfMention.statusConnection !== 'online') {
-				RocketChat.PushNotification.send({
-					roomId: message.rid,
-					username: push_username,
-					message: push_message,
-					badge: getBadgeCount(userOfMention._id),
-					payload: {
-						host: Meteor.absoluteUrl(),
-						rid: message.rid,
-						sender: message.u,
-						type: room.t,
-						name: room.name
-					},
-					usersTo: {
-						userId: userOfMention._id
-					}
-				});
-				return message;
+			if (canBeNotified(userOfMentionId, 'mobile')) {
+				if (Push.enabled === true && userOfMention.statusConnection !== 'online') {
```

Comment:
`busy` also receives notification?

Reply:
@rafaelks the behavior was not changed

Reply:
But that's a bug, when User is away it should receive notification on iOS as well.

Reply:
Its kind of confusing but there are two different status's.

Users Presence status (what they want to be perceived as) and their connection status.  

Their connection status only goes between basically 2 states.  online and away.  Offline would be that connection would not exist.

So if all connection status's are away the users status is automatically set to away which would allow them to receive push notifications to the phone.

Same with if all connections are offline (aka there aren't any connection) the users status should be offline and they will again receive notifications.